jerry-core/ecma/operations/ecma-typedarray-object.c

```diff
@@ -60,62 +51,61 @@ static ecma_number_t
 get_typedarray_element (lit_utf8_byte_t *src, /**< the location in the internal arraybuffer */
                         lit_magic_string_id_t class_id) /**< class name of the typedarray */
 {
-  ecma_number_t ret_num;
-
   switch (class_id)
   {
     case LIT_MAGIC_STRING_INT8_ARRAY_UL:
     {
-      GET_ELEMENT (int8_t, src, ret_num);
-      break;
+      return GET_ELEMENT (int8_t, src);
     }
     case LIT_MAGIC_STRING_UINT8_ARRAY_UL:
     case LIT_MAGIC_STRING_UINT8_CLAMPED_ARRAY_UL:
     {
-      GET_ELEMENT (uint8_t, src, ret_num);
-      break;
+      return GET_ELEMENT (uint8_t, src);
     }
     case LIT_MAGIC_STRING_INT16_ARRAY_UL:
     {
-      GET_ELEMENT (int16_t, src, ret_num);
-      break;
+      return GET_ELEMENT (int16_t, src);
     }
     case LIT_MAGIC_STRING_UINT16_ARRAY_UL:
     {
-      GET_ELEMENT (uint16_t, src, ret_num);
-      break;
+      return GET_ELEMENT (uint16_t, src);
     }
     case LIT_MAGIC_STRING_INT32_ARRAY_UL:
     {
-      GET_ELEMENT (int32_t, src, ret_num);
-      break;
+      return GET_ELEMENT (int32_t, src);
     }
     case LIT_MAGIC_STRING_UINT32_ARRAY_UL:
     {
-      GET_ELEMENT (uint32_t, src, ret_num);
-      break;
+      return GET_ELEMENT (uint32_t, src);
     }
     case LIT_MAGIC_STRING_FLOAT32_ARRAY_UL:
     {
-      GET_ELEMENT (float, src, ret_num);
-      break;
+      return GET_ELEMENT (float, src);
     }
 #if CONFIG_ECMA_NUMBER_TYPE == CONFIG_ECMA_NUMBER_FLOAT64
     case LIT_MAGIC_STRING_FLOAT64_ARRAY_UL:
     {
-      GET_ELEMENT (double, src, ret_num);
-      break;
+      return GET_ELEMENT (double, src);
     }
 #endif /* CONFIG_ECMA_NUMBER_TYPE == CONFIG_ECMA_NUMBER_FLOAT64 */
     default:
     {
       JERRY_UNREACHABLE ();
+      return 0;
     }
   }
-
-  return ret_num;
 } /* get_typedarray_element */
 
+#undef GET_ELEMENT
+
+/**
+ * Cast the number from ecma_numbet_t into type,
+ * and store the number in the location
+ */
+#define SET_ELEMENT(type, location, number) \
+  int32_t v = (int32_t) number; \
+  *((type *) location) = (type) v;
```

Comment:
I found the following:

 687 If the value of the integral part cannot be represented by the integer type, the behavior is undefined.50) 

http://c0x.coding-guidelines.com/6.3.1.4.html

I just realized that int32 and int64 conversions works differently on x86, and the standard reveals that most of these conversions are undefined, so we cannot rely on a specific behaviour.